src/services/codefixes/importFixes.ts

```diff
@@ -0,0 +1,393 @@
+﻿/* @internal */
+namespace ts.codefix {
+
+    registerCodeFix({
+        errorCodes: [Diagnostics.Cannot_find_name_0.code],
+        getCodeActions: (context: CodeFixContext) => {
+            const sourceFile = context.sourceFile;
+            const checker = context.program.getTypeChecker();
+            const allSourceFiles = context.program.getSourceFiles();
+            const useCaseSensitiveFileNames = context.host.useCaseSensitiveFileNames ? context.host.useCaseSensitiveFileNames() : false;
+
+            const token = getTokenAtPosition(sourceFile, context.span.start);
+            const name = token.getText();
+            const allActions: CodeAction[] = [];
+
+            const allPotentialModules = checker.getAmbientModules();
+            for (const otherSourceFile of allSourceFiles) {
+                if (otherSourceFile !== sourceFile && otherSourceFile.symbol) {
+                    allPotentialModules.push(otherSourceFile.symbol);
+                }
+            }
+
+            const currentTokenMeaning = getMeaningFromLocation(token);
+            for (const moduleSymbol of allPotentialModules) {
+                context.cancellationToken.throwIfCancellationRequested();
+
+                const moduleExports = checker.getExportsOfModule(moduleSymbol);
+                if (!moduleExports) {
+                    continue;
+                }
+
+                // check the default export
+                const defaultExport = moduleExports["default"];
+                if (defaultExport) {
+                    const localSymbol = getLocalSymbolForExportDefault(defaultExport);
+                    if (localSymbol && localSymbol.name === name && checkSymbolHasMeaning(localSymbol, currentTokenMeaning)) {
+                        allActions.push(getCodeActionForImport(moduleSymbol, /*isDefaultExport*/ true));
+                    }

```

Comment:
I don't understand what the above few lines do. Can you add some explanatory comments? In what scenario can we detect that you need to import a default export? It has no name to match on, as the local name for the 'default' export is provided by the import statement.